Make libmodulemd dependency conditional on WITH_MODULEMD option

Backport from upstream 5.2.x series.

--- a/libdnf5/CMakeLists.txt
+++ b/libdnf5/CMakeLists.txt
@@ -47,9 +47,11 @@ pkg_check_modules(JSONC REQUIRED json-c)
 include_directories(${JSONC_INCLUDE_DIRS})
 target_link_libraries(libdnf5 PRIVATE ${JSONC_LIBRARIES})
 
-pkg_check_modules(LIBMODULEMD REQUIRED modulemd-2.0>=2.11.2)
-list(APPEND LIBDNF5_PC_REQUIRES "${LIBMODULEMD_MODULE_NAME}")
-target_link_libraries(libdnf5 PRIVATE ${LIBMODULEMD_LIBRARIES})
+if(WITH_MODULEMD)
+    pkg_check_modules(LIBMODULEMD REQUIRED modulemd-2.0>=2.11.2)
+    list(APPEND LIBDNF5_PC_REQUIRES "${LIBMODULEMD_MODULE_NAME}")
+    target_link_libraries(libdnf5 PRIVATE ${LIBMODULEMD_LIBRARIES})
+endif()
 
 pkg_check_modules(LIBSOLV REQUIRED libsolv>=0.7.25)
 list(APPEND LIBDNF5_PC_REQUIRES "${LIBSOLV_MODULE_NAME}")
